File: BT_first_class/app/src/main/java/com/example/bt_first_class/MainActivity.java
package com.example.bt_first_class;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
     private RadioButton rdoBnt_nam,rdoBtn_nu;
     private EditText tx_age,tx_feet,tx_inches,tx_heigh;
     private Button btn_resuilt;
     private  TextView tx_BMI,tx_giatri;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        rdoBnt_nam = findViewById(R.id.rdButton_nam);
        rdoBtn_nu = findViewById(R.id.rdButton_nu);
        tx_age = findViewById(R.id.tx_age);
        tx_feet = findViewById(R.id.tx_feet);
        tx_inches = findViewById(R.id.tx_inches);
        tx_heigh = findViewById(R.id.tx_heigh);
        tx_BMI = findViewById(R.id.tx_BMI);
        tx_giatri = findViewById(R.id.tx_result);
        btn_resuilt = findViewById(R.id.btn_result);
        btn_resuilt.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this,"dfg", Toast.LENGTH_SHORT).show();
            }
        });

    }
}

File: sharePreference_TH1/app/src/main/java/com/example/sharepreference_th1/MainActivity.kt
package com.example.sharepreference_th1

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.ComponentActivity
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject


class MainActivity : ComponentActivity() {
    private lateinit var txtName : EditText
    private lateinit var txtPass : EditText
    private lateinit var txtShow : TextView
    private lateinit var btnAdd : Button
    private lateinit var btnShow : Button
    private lateinit var btnXoa : Button
    @SuppressLint("MissingInflatedId")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_activity)
        txtName=findViewById(R.id.txt_name)
        txtPass=findViewById(R.id.txt_pass)
        txtShow=findViewById(R.id.txt_show)
        btnXoa=findViewById(R.id.btn_xoa)
        btnAdd=findViewById(R.id.btn_them)
        btnShow=findViewById(R.id.btn_show)

        btnAdd.setOnClickListener{
            val tenUser = txtName.text.toString()
            val passUser = txtPass.text.toString()
            val infoUser = "$tenUser $passUser"
             PreferenceHelper(this).addUser(infoUser)

        }


        btnXoa.setOnClickListener{
           PreferenceHelper(this).deleteUser()
            txtShow.text = "Xóa thành công!"
        }
        btnShow.setOnClickListener{

            PreferenceHelper(this).showUser().let { result ->
                txtShow.text = result
            }

    }
}}

